src/NServiceBus.Metrics.ServiceControl.Tests/QueueLength/When_publishing_message.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using NServiceBus.AcceptanceTesting;
+using NServiceBus.Features;
+using NUnit.Framework;
+
+namespace NServiceBus.Metrics.ServiceControl.Tests
+{
+    using AcceptanceTests.PubSub;
+    using Config;
+    using Config.ConfigurationSource;
+    using EndpointTemplates;
+
+    public class When_publishing_message : NServiceBusAcceptanceTest
+    {
+        static Guid HostId = Guid.NewGuid();
+
+        [Test]
+        public void Should_enhance_it_with_queue_length_properties()
+        {
+            var context = Scenario.Define<Context>()
+                .WithEndpoint<Subscriber>(b => b.When(ctx => ctx.EndpointsStarted, (session, c) =>
+                {
+                    session.Subscribe<TestEventMessage1>();
+                    session.Subscribe<TestEventMessage2>();
+                }))
+                .WithEndpoint<Publisher>(c => c.When(ctx => ctx.SubscriptionCount == 2, s =>
+                {
+                    s.Publish(new TestEventMessage1());
+                    s.Publish(new TestEventMessage1());
+                    s.Publish(new TestEventMessage2());
+                    s.Publish(new TestEventMessage2());
+                }))
+                .WithEndpoint<MonitoringSpy>()
+                .Done(c => c.Headers1.Count == 2 && c.Headers2.Count == 2 && c.Data != null)
+                .Run();
+
+            AssertSequencesReported(context);
+        }
+
+        static void AssertSequencesReported(Context context)
+        {
+            var sessionIds = new[]
+            {
+            AssertHeaders(context.Headers1),
+            AssertHeaders(context.Headers2)
+        };
+
+            var data = context.Data;
+            var counters = data.Counters;
+            var counterTokens = counters.Where(c => c.Name.StartsWith("Sent sequence for"));
```

Comment:
Can we have counters that do not start with `Sent sequence for`?

Reply:
I think we should not care. This test was aimed to provide a contract that is consumed by SC.Monitoring. I'm fine with leaving it as it is, as it will be consumed by SC.Monitoring on the old basis.